.github/workflows/checks.yml

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2025 The Contributors to Eclipse OpenSOVD (see CONTRIBUTORS)
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: CI Checks
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+  workflow_call:
+    inputs:
+      rust-nightly-version:
+        description: 'Rust nightly version to use (YYYY-MM-DD)'
+        required: false
+        type: string
+      python-version:
+        description: 'Python version to use for pre-commit environment'
+        required: false
+        type: string
+      pre-commit-version:
+        description: 'Version of pre-commit to install'
+        required: false
+        type: string
+      pre-commit-config-path:
+        description: 'Path to a custom .pre-commit-config.yml in the consumer repository'
+        required: false
+        type: string
+      license-config-path:
+        description: 'Path to a custom .licenserc.yml in the consumer repository'
+        required: false
+        type: string
+
+permissions:
+  contents: read
+
+jobs:
+  checks:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run checks
+        uses: eclipse-opensovd/cicd-workflows/pre-commit-action@init
```

Comment:
must be init until we have this on main, immediately after merging this, we need a second PR changing this to main.